### PR TITLE
Indexing improvements

### PR DESCRIPTION
**Pull request by bjhargrave**

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java [hunk 1]
```diff
@@ -312,12 +316,12 @@ public String put(String key, String value) {
 						}
 						break;
 				}
-				return map.put(attribute, value);
+				return attribute;
 			}
 		}
 		// default String type
 		types.remove(key);
-		return map.put(key, value);
```
Comment (pkriens) [inline biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java]: Why did you change the return type?

File: biz.aQute.bndlib/src/aQute/bnd/print/JarPrinter.java [hunk 2]
```diff
@@ -286,25 +286,27 @@ public JarPrinter doManifest(Manifest manifest) {
...
-				sb.append(nibble(c));
+				sb.append('\\')
+					.append('u')
+					.append(nibble(c >> 12))
+					.append(nibble(c >> 8))
+					.append(nibble(c >> 4))
+					.append(nibble(c));
 			}
 		}
-		return sb.toString();
```
Comment (pkriens) [inline biz.aQute.bndlib/src/aQute/bnd/print/JarPrinter.java]: If you're going to uglify the code all over the place, should you not at least move this into a central method? Repeating this low level checks all over the place is error prone.

File: biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java (hunk 1 above)
Comment (bjhargrave) [reply biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java]: I split the method in two. One part handles computing the type and modifying the `type` map and the other part modifies the `map` map.
